File: quark/test/emit/map/js/map.js
var _qrt = require("quark_runtime.js");

function test1() {
    var map = new Map();
    (map).set(("pi"), (3));
    _qrt.print(_qrt.map_get((map), ("pi")));
    _qrt.print(_qrt.map_get((map), ("not_there")));
    _qrt.print("^--- should be null");
}
exports.test1 = test1;

function test_update() {
    var first = new Map();
    var second = new Map();
    (first).set(("a"), ("first_a"));
    (first).set(("b"), ("first_b"));
    (second).set(("b"), ("second_b"));
    (second).set(("c"), ("second_c"));
    (second).forEach(function (v, k) { (first).set(k, v); });
    _qrt.print(_qrt.map_get((first), ("a")));
    _qrt.print(_qrt.map_get((first), ("b")));
    _qrt.print(_qrt.map_get((first), ("c")));
}
exports.test_update = test_update;

function test_literal() {
    var map = new Map([]);
    map = new Map([["pi", 3.14159], ["e", 2.718]]);
    _qrt.print(_qrt.map_get((map), ("pi")));
    _qrt.print(_qrt.map_get((map), ("e")));
}
exports.test_literal = test_literal;

function main() {
    test1();
    test_update();
    test_literal();
}
exports.main = main;

main();

File: quarkc/test/emit/expected/rb/map/lib/map.rb
module Quark
require "quark"
def self.map; MODULE_map; end
module MODULE_map
require_relative 'map' # 0 () ()

def self.test1()
    
    map = ::Hash.new()
    (map)["pi"] = (3)
    ::DatawireQuarkCore.print((map)["pi"])
    ::DatawireQuarkCore.print((map)["not_there"])
    ::DatawireQuarkCore.print("^--- should be null")


    nil
end

def self.test_update()
    
    first = ::Hash.new()
    second = ::Hash.new()
    (first)["a"] = ("first_a")
    (first)["b"] = ("first_b")
    (second)["b"] = ("second_b")
    (second)["c"] = ("second_c")
    (first).merge!(second)
    ::DatawireQuarkCore.print((first)["a"])
    ::DatawireQuarkCore.print((first)["b"])
    ::DatawireQuarkCore.print((first)["c"])


    nil
end

def self.test_literal()
    
    map = {}
    map = {"pi" => 3.14159, "e" => 2.718}
    ::DatawireQuarkCore.print((map)["pi"])
    ::DatawireQuarkCore.print((map)["e"])


    nil
end

def self.iterables()
    
    numbers = {"zero" => 0, "one" => 1, "two" => 2, "e" => 2, "three" => 3, "pi" => 3, "fun" => 69, "dockingbay" => 94}
    keys = (numbers).keys
    (keys).sort!
    ::DatawireQuarkCore.print(keys)


    nil
end

def self.main()
    
    ::Quark.map.test1()
    ::Quark.map.test_update()
    ::Quark.map.test_literal()
    ::Quark.map.iterables()


    nil
end

if __FILE__ == $0
    ::Quark.map.main()
end

end # module MODULE_map
end # module Quark
